plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/server/commands/N4JSCommandService.java

```diff
@@ -10,41 +10,249 @@
+	}
+
+	/**
+	 * Clean the state of all workspace projects and trigger a re-initialization (new build).
+	 *
+	 * @param access
+	 *            the language server access.
+	 * @param cancelIndicator
+	 *            the cancel indicator.
+	 */
+	@ExecutableCommandHandler(N4JS_REBUILD)
+	public Void rebuild(ILanguageServerAccess access, CancelIndicator cancelIndicator) {
+		lspServer.clean();
+		lspServer.reinitWorkspace();
+		return null;
+	}
+
+	/**
+	 * Fix the issues of the same kind in the entire file.
+	 */
+	@ExecutableCommandHandler(COMPOSITE_FIX_FILE)
+	public Void fixAllInFile(String title, String code, String fixId, CodeActionParams codeActionParams,
+			ILanguageServerAccess access,
+			CancelIndicator cancelIndicator) {
+		WorkspaceEdit edit = codeActionService.applyToFile(code, fixId,
+				lspServer.toOptions(codeActionParams, cancelIndicator));
+		access.getLanguageClient().applyEdit(new ApplyWorkspaceEditParams(edit, title));
+		return null;
+	}
+
+	/**
+	 * Fix the issues of the same kind in the entire project.
+	 */
+	@ExecutableCommandHandler(COMPOSITE_FIX_PROJECT)
+	public Void fixAllInProject(String title, String code, String fixId, CodeActionParams codeActionParams,
+			ILanguageServerAccess access,
+			CancelIndicator cancelIndicator) {
+		WorkspaceEdit edit = codeActionService.applyToProject(code, fixId,
+				lspServer.toOptions(codeActionParams, cancelIndicator));
+		access.getLanguageClient().applyEdit(new ApplyWorkspaceEditParams(edit, title));
+		return null;
+	}
+
+	/**
+	 * Install the given NPM into the workspace.
+	 *
+	 * @param cancelIndicator
+	 *            not required.
+	 */
+	@ExecutableCommandHandler(JSONCodeActionService.INSTALL_NPM)
+	public Void installNpm(
+			String packageName,
+			String version,
+			String fileUri,
+			ILanguageServerAccess access,
+			CancelIndicator cancelIndicator) {
+		lspServer.getRequestManager().runWrite("InstallNpm", () -> {
+			NPMVersionRequirement versionRequirement = semverHelper.parse(version);
+			if (versionRequirement == null) {
+				versionRequirement = SemverUtils.createEmptyVersionRequirement();
+			}
+			LibraryChange change = new LibraryChange(Install, null, new N4JSProjectName(packageName),
+					SemverSerializer.serialize(versionRequirement));
+			MultiStatus multiStatus = new MultiStatus("json", 1, null, null);
+			npmCli.batchInstall(new NullProgressMonitor(), multiStatus,
+					Arrays.asList(change),
+					new FileURI(URI.createURI(fileUri)).getParent());
+
+			return multiStatus;
+		}, (ci, ms) -> {
+			MessageParams messageParams = new MessageParams();
+			switch (ms.getSeverity()) {
+			case IStatus.INFO:
+				messageParams.setType(MessageType.Info);
+				break;
+			case IStatus.WARNING:
+				messageParams.setType(MessageType.Warning);
+				break;
+			case IStatus.ERROR:
+				messageParams.setType(MessageType.Error);
+				break;
+			default:
+				return null;
+			}
+
+			StringWriter sw = new StringWriter();
```

Comment:
TODO message is ugly in VS code. Html did not work. Check what VSCode accepts to render a pretty message.